pkgs/applications/window-managers/sway/default.nix

```diff
@@ -8,8 +8,16 @@
 , isNixOS ? false
 
 , enableXWayland ? true
+, systemdSupport ? stdenv.isLinux
+, dbusSupport ? true
+, trayEnabled ? dbusSupport
 }:
 
+# sway's meson_options.txt does not include a "none" option for
+# "sd-bus".  However if you choose "basu" and disable tray support, it
+# will build without needing basu (neither at build time nor run time).
```

Comment:
Note: Not sure if that will be future proof but that doesn't mean that we cannot use that trick for now. I didn't change the source-code though (at least not yet). It would probably make sense to contact upstream regarding a `-Dsd-bus-provider=none` option or something similar (they might not like it though if they see it as a hard requirement - one should check first if this was already discussed).

Reply:
Done: https://github.com/swaywm/sway/issues/6843

Reply:
Hrm, well, the [reply](https://github.com/swaywm/sway/issues/6843#issuecomment-1047288761) was sort of "working as intended".

It looks like there are five possibilities not including `basu` which is not (currently) in nixpkgs:

1. libsystemd (and dbus) and tray=enabled
2. libsystemd (and dbus) and tray=disabled
3. tray=disabled
4. elogind (and dbus) and tray=enabled
5. elogind (and dbus) and tray=disabled

The last two possibilities are not (currently) supported by nixpkgs' sway derivation.

I have updated the PR to include a link to the sway maintainers' comment, rewritten the `-Dsd-bus-provider=` flag more compactly, and factored out a definition of `sd_bus_provider` so I could place that definition close to the `assert` that explains the whole situation up at the top of the file (otherwise somebody looking only at `mesonFlags` would think to themselves "what is this `basu` stuff?").

I think that if `elogindSupport` were to be added at some point in the future, the `assert` line would change from

```
assert trayEnabled -> systemdSupport && dbusSupport;
```

to

```
assert trayEnabled -> (systemdSupport || elogindSupport) && dbusSupport;
```

and an `else if` would be added to the `sd_bus_provider` definition.